src/app/file/webp_format.cpp

```diff
@@ -0,0 +1,361 @@
+  FILE* fp = handle.get();
+
+  struct writerData wd = {fp, fop};
+
+  auto image = fop->seq.image.get();
+  if (image->width() > WEBP_MAX_DIMENSION || image->height() > WEBP_MAX_DIMENSION) {
+   fop_error(
+     fop, "Error: WebP can only have a maximum width and height of %i but your %s has a size of %i x %i\n",
+     WEBP_MAX_DIMENSION, fop->filename.c_str(), image->width(), image->height()
+   );
+   return false;
+  }
+
+  base::SharedPtr<WebPOptions> webp_options = fop->seq.format_options;
+
+  WebPConfig config;
+
+  if (webp_options->lossless) {
+    if (!(WebPConfigInit(&config) && WebPConfigLosslessPreset(&config, webp_options->method))) {
+     fop_error(fop, "Error for WebP Config Version for file %s\n", fop->filename.c_str());
+     return false;
+    }
+    config.image_hint = webp_options->image_hint;
+  } else {
+    if (!WebPConfigPreset(&config, webp_options->image_preset, webp_options->quality)) {
+     fop_error(fop, "Error for WebP Config Version for file %s\n", fop->filename.c_str());
+     return false;
+    }
+  }
+
+  if (!WebPValidateConfig(&config)) {
+   fop_error(fop, "Error in WebP Encoder Config for file %s\n", fop->filename.c_str());
+   return false;
+  }
+
+  WebPPicture pic;
+  if (!WebPPictureInit(&pic)) {
+    fop_error(fop, "Error for WebP Picture Version mismatch for file %s\n", fop->filename.c_str());
+    return false;
+  }
+
+  pic.width = image->width();
+  pic.height = image->height();
+  if (webp_options->lossless) {
+    pic.use_argb = true;
+  }
+
+  if (!WebPPictureAlloc(&pic)) {
+    fop_error(fop, "Error for  WebP Picture memory allocations for file %s\n", fop->filename.c_str());
+    return false;
+  }
+
+  if (!WebPPictureImportRGBA(&pic, (uint8_t*)image->getPixelAddress(0, 0), image->width() * sizeof(uint32_t))) {
+    fop_error(fop, "Error for LibWebP Import RGBA Buffer into Picture for %s\n", fop->filename.c_str());
+    WebPPictureFree(&pic);
+    return false;
+  }
+
+  pic.writer = FileWriter;
+  pic.custom_ptr = &wd;
+  pic.progress_hook = ProgressReport;
+
+  if (!WebPEncode(&config, &pic)) {
+    fop_error(fop, "Error for LibWebP while Encoding %s: %s\n", fop->filename.c_str(), WebPEncodingErrorMap.find(pic.error_code)->second.c_str());
+    WebPPictureFree(&pic);
+    return false;
+  }
+
+  WebPPictureFree(&pic);
+  return true;
+}
+#endif
+
+// Shows the WebP configuration dialog.
+base::SharedPtr<FormatOptions> WebPFormat::onGetFormatOptions(FileOp* fop)
+{
+  base::SharedPtr<WebPOptions> webp_options;
+  if (fop->document->getFormatOptions())
+    webp_options = base::SharedPtr<WebPOptions>(fop->document->getFormatOptions());
+
+  if (!webp_options)
+    webp_options.reset(new WebPOptions);
+
+  // Non-interactive mode
+  if (!fop->context || !fop->context->isUIAvailable())
+    return webp_options;
+
+  try {
+
+    // Load the window to ask to the user the JPEG options he wants.
+    UniquePtr<ui::Window> window(app::load_widget<ui::Window>("webp_options.xml", "webp_options"));
```

Comment:
You can use the auto-generated widget here. The .h is generated in `your_build_dir/src/app/generated_webp_options.h` and you can use it in this way:
```
#include "generated_webp_options.h"
...
app::gen::WebpOptions window;
...
```
You can find [here](https://github.com/aseprite/aseprite/blob/master/src/app/file/gif_format.cpp#L1247) an example.